reference/reference.go

```diff
@@ -236,6 +237,18 @@ func WithDigest(name Named, digest digest.Digest) (Canonical, error) {
 	}, nil
 }
 
+// Match reports whether ref matches the specified pattern.
+func Match(pattern string, ref Reference) bool {
+	matched, err := path.Match(pattern, ref.String())
+	if err != nil {
+		return false
+	}
+	if namedRef, isNamed := ref.(Named); isNamed && !matched {
+		matched, _ = path.Match(pattern, namedRef.Name())
```

Comment:
This is returning `matches(ref.String()) || matches(ref.Name())` if the reference is `Named`, others ref will only check `ref.String()`. The basic idea is to map a little bit what `docker images` did (and a little more). That means `foo/*/ba[rz]` would match `foo/a/bar:tag` for example. To achieve that, I need to check if the name part matches if the whole string does not (this way `foo/*/ba[rz]:t*` will match `foo/a/bar:tag` too).

The error is not checked here because it cannot happen the only error is if the pattern is wrong, and this error is already checked just before 👼.